vela-templates/definitions/api-oriented/component/webservice.cue

```diff
@@ -0,0 +1,260 @@
+			template: {
+				metadata: labels: {
+					"app.oam.dev/component": context.name
+					if parameter.addRevisionLabel {
+						"app.oam.dev/appRevision": context.appRevision
+					}
+					"app.oam.dev/revision": context.revision
+				}
+
+				spec: {
+					containers: [{
+						name:  context.name
+						image: parameter.image
+						if parameter["ports"] == _|_ {
+							ports: [{
+								containerPort: 80
+							}]
+						}
+						if parameter["ports"] != _|_ {
+							ports: [ for v in parameter.ports {
+								{
+									containerPort: v
+								}}]
+						}
+
+						if parameter["imagePullPolicy"] != _|_ {
+							imagePullPolicy: parameter.imagePullPolicy
+						}
+
+						if parameter["cmd"] != _|_ {
+							command: parameter.cmd
+						}
+
+						if parameter["env"] != _|_ {
+							env: parameter.env
+						}
+
+						if context["config"] != _|_ {
+							env: context.config
+						}
+
+						if parameter["cpu"] != _|_ {
+							resources: {
+								limits: cpu:   parameter.cpu
+								requests: cpu: parameter.cpu
+							}
+						}
+
+						if parameter["memory"] != _|_ {
+							resources: {
+								limits: memory:   parameter.memory
+								requests: memory: parameter.memory
+							}
+						}
+
+						if parameter["volumes"] != _|_ {
+							volumeMounts: [ for v in parameter.volumes {
+								{
+									mountPath: v.mountPath
+									name:      v.name
+								}}]
+						}
+
+						if parameter["livenessProbe"] != _|_ {
+							livenessProbe: parameter.livenessProbe
+						}
+
+						if parameter["readinessProbe"] != _|_ {
+							readinessProbe: parameter.readinessProbe
+						}
+
+					}]
+
+					if parameter["imagePullSecrets"] != _|_ {
+						imagePullSecrets: [ for v in parameter.imagePullSecrets {
+							name: v
+						},
+						]
+					}
+
+					if parameter["volumes"] != _|_ {
+						volumes: [ for v in parameter.volumes {
+							{
+								name: v.name
+								if v.type == "pvc" {
+									persistentVolumeClaim: claimName: v.claimName
+								}
+								if v.type == "configMap" {
+									configMap: {
+										defaultMode: v.defaultMode
+										name:        v.cmName
+										if v.items != _|_ {
+											items: v.items
+										}
+									}
+								}
+								if v.type == "secret" {
+									secret: {
+										defaultMode: v.defaultMode
+										secretName:  v.secretName
+										if v.items != _|_ {
+											items: v.items
+										}
+									}
+								}
+								if v.type == "emptyDir" {
+									emptyDir: medium: v.medium
+								}
+							}}]
+					}
+				}
+			}
+		}
+	}
+	parameter: {
+		// +usage=Which image would you like to use for your service
+		// +short=i
+		image: string
+
+		// +usage=Specify image pull policy for your service
+		imagePullPolicy?: string
+
+		// +usage=Specify image pull secrets for your service
+		imagePullSecrets?: [...string]
+
+		// +usage=Which ports do you want customer traffic sent to, defaults to 80
+		// +short=p
+		ports?: [...int]
```

Comment:
did support set port number, name, and protocol?

Reply:
Is the name necessary to be exposed to users?